libs/frontend/util-edit-text/src/lib/text-to-link-keymap.ts

```diff
@@ -0,0 +1,103 @@
+import { Node, Schema, ResolvedPos } from 'prosemirror-model'
+import { Command, EditorState, TextSelection, Transaction } from 'prosemirror-state'
+import { createLinkNode } from './prosemirror-utils'
+import {
+  baseKeymap,
+  chainCommands,
+  createParagraphNear,
+  liftEmptyBlock,
+  splitBlock,
+} from './base-keymap'
+
+export const urlRegex = () =>
+  /(https?:\/\/(www\.)?)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\\+.~#?&/=]*)/g
+
+export const textToLinkKeymap = (schema: Schema) => {
+  const keys: Record<string, Command> = {}
+
+  keys.Space = textToLinkOnSpace(schema)
+  keys.Enter = textToLinkOnEnter(schema)
+
+  return keys
+}
+
+const textToLinkOnSpace = (schema: Schema): Command => {
+  return (state, dispatch) => {
+    const { $cursor } = state.selection as TextSelection
+    if ($cursor) {
+      let transaction = convertTextToLinkWhenNecessary(schema, state, $cursor)
+      if (transaction) {
+        // Manually insert a space character because underline mark will be applied to
+        // the default space character. We can avoid this by creating a new text node
+        // without any mark.
+        const spaceNode = schema.text(' ')
+        transaction = transaction.insert(state.selection.head, spaceNode)
+        dispatch?.(transaction)
+        return true
+      }
+    }
+    return false
+  }
+}
+
+const textToLinkOnEnter = (schema: Schema): Command => {
+  return (state, dispatch, view) => {
+    const { $cursor } = state.selection as TextSelection
+    if ($cursor) {
+      const transaction = convertTextToLinkWhenNecessary(schema, state, $cursor)
+      if (transaction) {
+        const newState = state.apply(transaction)
+        if (view) {
+          view.updateState(newState)
+          // Pass the updated state to base command so that transaction could work properly.
+          baseKeymap.Enter?.(newState, dispatch, view)
+        }
+        return true
+      } else {
+        return chainCommands(createParagraphNear, liftEmptyBlock, splitBlock)(
+          state,
+          dispatch,
+          view,
+        )
```

Comment:
The base `Enter` keymap is `chainCommands(newlineInCode, createParagraphNear, liftEmptyBlock, splitBlock)`, but since this plugin is only used in ProseMirrorEditor for component, I removed `newlineInCode` here and use the commands from the copied version of `baseKeymap` for #175 .